blocks/gu_spoverview/tests/spoverview_test.php

```diff
@@ -0,0 +1,251 @@
+        $assignments_overdue = 0;
+        
+        $courses = enrol_get_all_users_courses($this->student->id, true);
+        $courseids = array_column($courses, 'id');
+
+        $assignments = (count($courseids) > 0) ? self::get_user_assignments($this->student->id, $courseids) : 0;
+
+        foreach ($assignments as $assignment) {
+            if($assignment->status != 'submitted') {
+                if(time() > $assignment->startdate) {
+                    if(($assignment->duedate > 0 && time() <= $assignment->duedate)
+                        || ($assignment->cutoffdate > 0 && time() <= $assignment->cutoffdate)
+                        || (($assignment->extensionduedate > 0 || !is_null($assignment->extensionduedate))
+                            && time() <= $assignment->extensionduedate)) {
+                        $assignments_tosubmit++;
+                    }else{
+                        if($assignment->duedate != 0 || $assignment->cutoffdate != 0
+                            || $assignment->extensionduedate != 0) {
+                            $assignments_overdue++;
+                        }
+                    }
+                }
+            }else{
+                $assignments_submitted++;
+            }
+        }
+
+        $this->assertNotEmpty($courses, 'empty');
+        $this->assertNotEmpty($assignments, 'empty');
+
+        $this->assertGreaterThanOrEqual(1, $assignments_tosubmit);
+        $this->assertGreaterThanOrEqual(2, $assignments_overdue);
+        $this->assertGreaterThanOrEqual(0, $assignments_submitted);
+
+        $this->assertContains(
+            $this->course->id, 
+            $courseids, 
+            "courseids array doesn't contains same course id"
+        );
+    }
+
+    public function test_check_marked_assignment(){
+        $courses = enrol_get_all_users_courses($this->student->id, true);
+        $courseids = array_column($courses, 'id');
+        $assessments_marked = (count($courseids) > 0) ? self::get_user_assessments_count($this->student->id, $courseids) : 0;
+
+        $this->assertNotEmpty($courses, 'empty');
+        $this->assertGreaterThanOrEqual(0, $assessments_marked);
+
+        $this->assertContains(
+            $this->course->id, 
+            $courseids, 
+            "courseids array doesn't contains same course id"
+        );
+    }
+
+    public function test_applicable_formats(){
+        $returned = $this->spoverview->applicable_formats();
+
+        $this->assertEquals($returned, array('my' => true));
+    }
+
+    public function test_get_content(){
+        $lang = 'block_gu_spoverview';
+
+        $assignments_submitted = 0;
+        $assignments_tosubmit = 0;
+        $assignments_overdue = 0;
+        
+        $courses = enrol_get_all_users_courses($this->student->id, true);
+        $courseids = array_column($courses, 'id');
+
+        $assignments = (count($courseids) > 0) ? self::get_user_assignments($this->student->id, $courseids) : 0;
```

Comment:
Add $this->spoverview = new block_gu_spoverview(); and use $this->spoverview to call get_user_assignments.